cmd/tiltdemo/main.go

```diff
@@ -8,27 +8,101 @@ import (
 	"time"
 )
 
+var cGreen = "\033[32m"
+var cBlue = "\033[34m"
+var cReset = "\u001b[0m"
+
 func main() {
-	println(0, "Starting tilt...")
-	println(100, "Building [my-app-frontend]")
-	println(500, "Build complete in 3.3423s")
+	printlnColor(0, "Starting tilt!", cGreen)
+	println(20, "  → in context [docker-for-desktop]")
 
-	awaitInput()
+	newline()
+	printlnColor(100, "Building from scratch: [gcr.io/project/app-frontend]", cGreen)
+	println(100, "  → tarring context")
+	println(100, "  → sending to Docker daemon…")
+
+	newline()
+	printDockerBuild(3)
 
-	println(100, "Building [my-app-backend]")
-	println(500, "Build complete in 0.1232s")
+	newline()
+	println(100, "  → tagged 2917b4065035")
+	printlnColor(200, "  → Done in 7.234s", cBlue)
 
+	newline()
+	printlnColor(100, "Deploying: [devel-hanyu-lb-blorg-be]", cGreen)
+	println(100, "  → parsing config YAML")
+	println(100, "  → applying config via kubectl")
+	println(100, "  → service created")
+	println(100, "  → deployment created")
+	printlnColor(200, "  → Done in 472ms", cBlue)
+
+	newline()
+	printlnColor(100, "Awaiting your edits…", cGreen)
 	awaitInput()
+	printlnColor(200, "  → File edited: main.go:123", cBlue)
+
+	newline()
+	printlnColor(100, "Building from existing: [gcr.io/project/app-frontend]", cGreen)
+	println(100, "  → tarring context")
+	println(100, "  → sending to Docker daemon…")
+
+	newline()
+	printDockerBuild(1)
+
+	newline()
+	println(100, "  → tagged 2917b4065035")
+	printlnColor(200, "  → Done in 1.234s", cBlue)
+
+	newline()
+	printlnColor(100, "Deploying: [devel-hanyu-lb-blorg-be]", cGreen)
+	println(100, "  → parsing config YAML")
+	println(100, "  → applying config via kubectl")
+	println(100, "  → service created")
+	println(100, "  → deployment created")
+	printlnColor(200, "  → Done in 472ms", cBlue)
+	newline()
```

Comment:
I think my big remaining question is about terminology and timing for the overall pipeline run.

A user should see how long the entire thing took, and how much of that was their build (which we don't control), as a summary, without having to scroll.